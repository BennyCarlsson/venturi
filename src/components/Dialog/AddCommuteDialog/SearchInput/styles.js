import styled from 'styled-components';

export const Input = styled.input`
  width: 100%;
  height: 36px;
  border: none;
  border-bottom: 1px solid #726a95;
  font-family: 'Montserrat';
  font-style: normal;
  font-size: 14px;
  line-height: 17px;
  padding: 8px;
`;
