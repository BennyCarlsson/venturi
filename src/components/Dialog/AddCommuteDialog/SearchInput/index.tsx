import { Dispatch, SetStateAction } from "react";
import { Input } from "./styles";

type SearchInputProps = {
  value: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  setShowDropDown: Dispatch<SetStateAction<boolean>>;
};

const SearchInput = ({
  value,
  onChange,
  setShowDropDown,
  ...rest
}: SearchInputProps) => {
  return (
    <Input
      {...rest}
      value={value}
      onChange={onChange}
      onBlur={() => {
        setTimeout(() => {
          setShowDropDown(false);
        }, 100);
      }}
      onFocus={() => {
        if (value.length > 1) {
          setShowDropDown(true);
        }
      }}
    />
  );
};

export default SearchInput;
