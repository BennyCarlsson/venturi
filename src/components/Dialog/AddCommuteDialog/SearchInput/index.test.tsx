import { cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import SearchInput from '.';

describe('<SearchInput />', () => {
  const value = 'texttext';
  const onChange = jest.fn();
  const setShowDropDown = jest.fn();

  beforeEach(() => {
    render(<SearchInput value={value} onChange={onChange} setShowDropDown={setShowDropDown} />);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(
      <SearchInput value="123" onChange={jest.fn()} setShowDropDown={jest.fn()} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    expect(screen).toBeTruthy();
  });

  it('should have value as input value', () => {
    expect(screen.getByRole('textbox')).toHaveValue(value);
  });

  it('should call onChange onChange', () => {
    userEvent.type(screen.getByRole('textbox'), 'Hello');
    expect(onChange).toHaveBeenCalledTimes(5);
  });

  it('should call setShowDropDown onFocus', () => {
    screen.getByRole('textbox').focus();
    expect(setShowDropDown).toHaveBeenCalledTimes(1);
    expect(setShowDropDown).toHaveBeenCalledWith(true);
  });

  it('should not call setShowDropDown onFocus when value is none', () => {
    cleanup();
    render(<SearchInput value={''} onChange={onChange} setShowDropDown={setShowDropDown} />);
    screen.getByRole('textbox').focus();
    expect(setShowDropDown).not.toHaveBeenCalled();
  });

  it('should not call setShowDropDown onFocus with value.length is 1', () => {
    cleanup();
    render(<SearchInput value={'a'} onChange={onChange} setShowDropDown={setShowDropDown} />);
    screen.getByRole('textbox').focus();
    expect(setShowDropDown).not.toHaveBeenCalled();
  });

  it('should call setShowDropDown onBlur', async () => {
    screen.getByRole('textbox').focus();
    screen.getByRole('textbox').blur();

    expect(setShowDropDown).toHaveBeenCalledTimes(1); // from onFocus

    await waitFor(() => expect(setShowDropDown).toHaveBeenCalledTimes(2)); // from onBlur
    await waitFor(() => expect(setShowDropDown).toHaveBeenNthCalledWith(2, false));
  });
});
