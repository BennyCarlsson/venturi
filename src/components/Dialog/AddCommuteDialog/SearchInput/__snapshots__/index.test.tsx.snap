// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<input
  className="sc-bdfBwQ ebCodQ"
  onBlur={[Function]}
  onChange={[MockFunction]}
  onFocus={[Function]}
  value="123"
/>
`;
